The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15331284425/job/43138485232) [(plain)](https://github.com/rust-lang/rust/commit/74b1eb30284f3a835853c3f7b3bbabf164173291/checks/43138485232/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
71 warning: unresolved link to `BarD`
-   --> $DIR/warning.rs:45:20
+   --> $DIR/warning.rs:45:9
73    |
74 LL | #[doc = "Foo\nbar [BarD] bar\nbaz"]
-    |                    ^^^^ no item named `BarD` in scope
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
76    |
+    = note: the link appears in this line:
+            
+            bar [BarD] bar
+                 ^^^^
+    = note: no item named `BarD` in scope
77    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
78 
79 warning: unresolved link to `BarF`

-   --> $DIR/warning.rs:54:15
+   --> $DIR/warning.rs:54:4
81    |
82 LL | f!("Foo\nbar [BarF] bar\nbaz");
-    |               ^^^^ no item named `BarF` in scope
+    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^
84    |
+    = note: the link appears in this line:
+            
+            bar [BarF] bar
+                 ^^^^
+    = note: no item named `BarF` in scope
85    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
86    = note: this warning originates in the macro `f` (in Nightly builds, run with -Z macro-backtrace for more info)
87 

102    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
103 
104 warning: unresolved link to `error`
-   --> $DIR/warning.rs:68:23
+   --> $DIR/warning.rs:68:9
106    |
107 LL | #[doc = "single line [error]"]
-    |                       ^^^^^ no item named `error` in scope
+    |         ^^^^^^^^^^^^^^^^^^^^^
109    |
+    = note: the link appears in this line:
+            
---

-   --> $DIR/warning.rs:71:41
+   --> $DIR/warning.rs:71:9
114    |
115 LL | #[doc = "single line with \"escaping\" [error]"]
-    |                                         ^^^^^ no item named `error` in scope
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
117    |
+    = note: the link appears in this line:
+            
+            single line with "escaping" [error]
+                                         ^^^^^
+    = note: no item named `error` in scope
118    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
119 
120 warning: unresolved link to `error`
---
+   --> $DIR/warning.rs:45:9
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    = note: the link appears in this line:
+            
+            bar [BarD] bar
+                 ^^^^
+    = note: no item named `BarD` in scope
+   --> $DIR/warning.rs:54:4
+    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    = note: the link appears in this line:
+            
+            bar [BarF] bar
+                 ^^^^
+    = note: no item named `BarF` in scope
+   --> $DIR/warning.rs:68:9
+    |         ^^^^^^^^^^^^^^^^^^^^^
+    = note: the link appears in this line:
---
+   --> $DIR/warning.rs:71:9
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    = note: the link appears in this line:
+            
+            single line with "escaping" [error]
+                                         ^^^^^
+    = note: no item named `error` in scope


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args intra-doc/warning.rs`

error: 1 errors occurred comparing output.
status: exit status: 0
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/tests/rustdoc-ui/intra-doc/warning.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/rustdoc-ui/intra-doc/warning" "-A" "internal_features" "-Cdebuginfo=0"
stdout: none
--- stderr -------------------------------
warning: unresolved link to `Foo::baz`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:3:23
   |
LL |        //! Test with [Foo::baz], [Bar::foo], ...
   |                       ^^^^^^^^ the struct `Foo` has no field or associated item named `baz`
   |
   = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: unresolved link to `Bar::foo`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:3:35
   |
LL |        //! Test with [Foo::baz], [Bar::foo], ...
   |                                   ^^^^^^^^ no item named `Bar` in scope

warning: unresolved link to `Uniooon::X`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:6:13
   |
LL |      //! , [Uniooon::X] and [Qux::Z].
   |             ^^^^^^^^^^ no item named `Uniooon` in scope

warning: unresolved link to `Qux::Z`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:6:30
   |
LL |      //! , [Uniooon::X] and [Qux::Z].
   |                              ^^^^^^ no item named `Qux` in scope

warning: unresolved link to `Uniooon::X`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:10:14
   |
LL |       //! , [Uniooon::X] and [Qux::Z].
   |              ^^^^^^^^^^ no item named `Uniooon` in scope

warning: unresolved link to `Qux::Z`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:10:31
   |
LL |       //! , [Uniooon::X] and [Qux::Z].
   |                               ^^^^^^ no item named `Qux` in scope

warning: unresolved link to `Qux:Y`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:14:13
   |
LL |        /// [Qux:Y]
   |             ^^^^^ no item named `Qux:Y` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `BarA`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:21:10
   |
LL | /// bar [BarA] bar //~ WARNING `BarA`
   |          ^^^^ no item named `BarA` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `BarB`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:27:9
   |
LL |  * bar [BarB] bar //~ WARNING `BarB`
   |         ^^^^ no item named `BarB` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `BarC`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:34:6
   |
LL | bar [BarC] bar //~ WARNING `BarC`
   |      ^^^^ no item named `BarC` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `BarD`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:45:9
   |
LL | #[doc = "Foo\nbar [BarD] bar\nbaz"] //~ WARNING `BarD`
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the link appears in this line:
           
           bar [BarD] bar
                ^^^^
   = note: no item named `BarD` in scope
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `BarF`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:54:4
   |
LL | f!("Foo\nbar [BarF] bar\nbaz"); //~ WARNING `BarF`
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the link appears in this line:
           
           bar [BarF] bar
                ^^^^
   = note: no item named `BarF` in scope
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
   = note: this warning originates in the macro `f` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unresolved link to `error`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:58:30
   |
LL |  * time to introduce a link [error]*/ //~ WARNING `error`
   |                              ^^^^^ no item named `error` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:64:30
   |
LL |  * time to introduce a link [error] //~ WARNING `error`
   |                              ^^^^^ no item named `error` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:68:9
   |
LL | #[doc = "single line [error]"] //~ WARNING `error`
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the link appears in this line:
           
           single line [error]
                        ^^^^^
   = note: no item named `error` in scope
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:71:9
   |
LL | #[doc = "single line with \"escaping\" [error]"] //~ WARNING `error`
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the link appears in this line:
           
           single line with "escaping" [error]
                                        ^^^^^
   = note: no item named `error` in scope
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:74:1
   |
LL | / /// Item docs. //~ WARNING `error`
LL | | #[doc="Hello there!"]
LL | | /// [error]
   | |___________^
   |
   = note: the link appears in this line:
           
           [error]
            ^^^^^
   = note: no item named `error` in scope
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error1`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:80:11
   |
LL | /// docs [error1] //~ WARNING `error1`
   |           ^^^^^^ no item named `error1` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: unresolved link to `error2`
##[warning]  --> /checkout/tests/rustdoc-ui/intra-doc/warning.rs:82:11
   |
LL | /// docs [error2] //~ WARNING `error2`
   |           ^^^^^^ no item named `error2` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: 19 warnings emitted
---
210 error: this URL is not a hyperlink
-   --> $DIR/bare-urls.rs:41:26
+   --> $DIR/bare-urls.rs:41:9
212    |
213 LL | #[doc = "here's a thing: https://example.com/"]
-    |                          ^^^^^^^^^^^^^^^^^^^^
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
215    |
216    = note: bare URLs are not automatically turned into clickable links
- help: use an automatic link instead
-    |
- LL | #[doc = "here's a thing: <https://example.com/>"]
-    |                          +                    +
221 
222 error: this URL is not a hyperlink
-   --> $DIR/bare-urls.rs:45:5
+   --> $DIR/bare-urls.rs:45:1
224    |
- LL | /// https://example.com/sugar
-    |     ^^^^^^^^^^^^^^^^^^^^^^^^^
+ LL | / /// https://example.com/sugar
+ LL | |
+ LL | | #[doc = "https://example.com/raw"]
+    | |_________________________________^
227    |
228    = note: bare URLs are not automatically turned into clickable links
- help: use an automatic link instead
-    |
- LL | /// <https://example.com/sugar>
-    |     +                         +
233 
234 error: this URL is not a hyperlink
-   --> $DIR/bare-urls.rs:47:10
+   --> $DIR/bare-urls.rs:45:1
236    |
- LL | #[doc = "https://example.com/raw"]
-    |          ^^^^^^^^^^^^^^^^^^^^^^^
+ LL | / /// https://example.com/sugar
+ LL | |
+ LL | | #[doc = "https://example.com/raw"]
+    | |_________________________________^
239    |
240    = note: bare URLs are not automatically turned into clickable links
- help: use an automatic link instead
-    |
- LL | #[doc = "<https://example.com/raw>"]
-    |          +                       +
+    = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
245 
246 error: aborting due to 20 previous errors
247 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:41:9
-   --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:45:1
- LL | | //~^ ERROR this URL is not a hyperlink
-   --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:45:1
- LL | | //~^ ERROR this URL is not a hyperlink
+   --> $DIR/bare-urls.rs:41:9
+    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   --> $DIR/bare-urls.rs:45:1
+ LL | / /// https://example.com/sugar
+ LL | |
+ LL | | #[doc = "https://example.com/raw"]
+    | |_________________________________^
+   --> $DIR/bare-urls.rs:45:1
+ LL | / /// https://example.com/sugar
+ LL | |
+ LL | | #[doc = "https://example.com/raw"]
+    | |_________________________________^
+    = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`


The actual stderr differed from the expected stderr
Saved the actual fixed to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/rustdoc-ui/lints/bare-urls/bare-urls.fixed`
diff of fixed:

38 //~^ ERROR this URL is not a hyperlink
39 pub fn c() {}
40 
- #[doc = "here's a thing: <https://example.com/>"]
+ #[doc = "here's a thing: https://example.com/"]
42 //~^ ERROR this URL is not a hyperlink
43 pub fn f() {}
44 

- /// <https://example.com/sugar>
+ /// https://example.com/sugar
46 //~^ ERROR this URL is not a hyperlink
- #[doc = "<https://example.com/raw>"]
+ #[doc = "https://example.com/raw"]
48 //~^ ERROR this URL is not a hyperlink
49 pub fn mixed() {}
50 


The actual fixed differed from the expected fixed
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args lints/bare-urls.rs`

error: 2 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/tests/rustdoc-ui/lints/bare-urls.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/rustdoc-ui/lints/bare-urls" "-A" "internal_features" "-Cdebuginfo=0"
stdout: none
--- stderr -------------------------------
error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:5:5
   |
---
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://www.somewhere.com>
   |     +                         +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:11:5
   |
LL | /// https://www.somewhere.com/a
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://www.somewhere.com/a>
   |     +                           +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:13:5
   |
LL | /// https://subdomain.example.com
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://subdomain.example.com>
   |     +                             +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:15:5
   |
LL | /// https://somewhere.com?
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com?>
   |     +                      +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:17:5
   |
LL | /// https://somewhere.com/a?
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com/a?>
   |     +                        +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:19:5
   |
LL | /// https://somewhere.com?hello=12
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com?hello=12>
   |     +                              +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:21:5
   |
LL | /// https://somewhere.com/a?hello=12
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com/a?hello=12>
   |     +                                +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:23:5
   |
LL | /// https://example.com?hello=12#xyz
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://example.com?hello=12#xyz>
   |     +                                +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:25:5
   |
LL | /// https://example.com/a?hello=12#xyz
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://example.com/a?hello=12#xyz>
   |     +                                  +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:27:5
   |
LL | /// https://example.com#xyz
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://example.com#xyz>
   |     +                       +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:29:5
   |
LL | /// https://example.com/a#xyz
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://example.com/a#xyz>
   |     +                         +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:31:5
   |
LL | /// https://somewhere.com?hello=12&bye=11
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com?hello=12&bye=11>
   |     +                                     +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:33:5
   |
LL | /// https://somewhere.com/a?hello=12&bye=11
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com/a?hello=12&bye=11>
   |     +                                       +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:35:5
   |
LL | /// https://somewhere.com?hello=12&bye=11#xyz
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// <https://somewhere.com?hello=12&bye=11#xyz>
   |     +                                         +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:37:10
   |
LL | /// hey! https://somewhere.com/a?hello=12&bye=11#xyz
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links
help: use an automatic link instead
   |
LL | /// hey! <https://somewhere.com/a?hello=12&bye=11#xyz>
   |          +                                           +

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:41:9
   |
LL | #[doc = "here's a thing: https://example.com/"]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: bare URLs are not automatically turned into clickable links

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:45:1
   |
LL | / /// https://example.com/sugar
LL | | //~^ ERROR this URL is not a hyperlink
LL | | #[doc = "https://example.com/raw"]
   | |_________________________________^
   |
   = note: bare URLs are not automatically turned into clickable links

error: this URL is not a hyperlink
##[error]  --> /checkout/tests/rustdoc-ui/lints/bare-urls.rs:45:1
   |
LL | / /// https://example.com/sugar
LL | | //~^ ERROR this URL is not a hyperlink
LL | | #[doc = "https://example.com/raw"]
   | |_________________________________^
   |
   = note: bare URLs are not automatically turned into clickable links
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: aborting due to 20 previous errors
------------------------------------------


---
630 error: unescaped backtick
-   --> $DIR/unescaped_backticks.rs:108:10
+   --> $DIR/unescaped_backticks.rs:108:9
632    |
633 LL | #[doc = "`"]
-    |          ^
+    |         ^^^
635    |
636    = help: the opening or closing backtick of an inline code may be missing
637    = help: if you meant to use a literal backtick, escape it

639            to this: \`
640 
641 error: unescaped backtick
-   --> $DIR/unescaped_backticks.rs:115:26
+   --> $DIR/unescaped_backticks.rs:115:9
643    |
644 LL | #[doc = concat!("\\", "`")]
-    |                          ^
+    |         ^^^^^^^^^^^^^^^^^^^^
646    |
647    = help: the opening backtick of an inline code may be missing
648             change: \`
---
To only update this specific test, also pass `--test-args unescaped_backticks.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustdoc" "/checkout/tests/rustdoc-ui/unescaped_backticks.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/rustdoc-ui/unescaped_backticks" "-A" "internal_features" "-Cdebuginfo=0"
stdout: none
--- stderr -------------------------------
error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:187:70
   |
LL | /// if you want your MIR to be modified by the full MIR pipeline, or `#![custom_mir(dialect =
   |                                                                      ^
   |
note: the lint level is defined here
  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:1:9
   |
LL | #![deny(rustdoc::unescaped_backticks)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: the closing backtick of an inline code may be missing
   |
LL | /// "runtime", phase = "optimized")]` if you don't.
   |                                     +
help: if you meant to use a literal backtick, escape it
   |
LL | /// if you want your MIR to be modified by the full MIR pipeline, or \`#![custom_mir(dialect =
   |                                                                      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:232:13
   |
LL |         //! `#![rustc_expected_cgu_reuse(module="spike", cfg="rpass2", kind="post-lto")]
   |             ^
   |
help: the closing backtick of an inline code may be missing
   |
LL |         //! `#![rustc_expected_cgu_reuse(module="spike", cfg="rpass2", kind="post-lto")]`
   |                                                                                         +
help: if you meant to use a literal backtick, escape it
   |
LL |         //! \`#![rustc_expected_cgu_reuse(module="spike", cfg="rpass2", kind="post-lto")]
   |             +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:237:13
   |
LL |         /// `cfg=...
   |             ^
   |
help: the closing backtick of an inline code may be missing
   |
LL |         /// `cfg=...`
   |                     +
help: if you meant to use a literal backtick, escape it
   |
LL |         /// \`cfg=...
   |             +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:241:42
   |
LL |         /// `cfg=... and not `#[cfg_attr]`
   |                                          ^
   |
help: a previous inline code might be longer than expected
   |
LL |         /// `cfg=...` and not `#[cfg_attr]`
   |                     +
help: if you meant to use a literal backtick, escape it
   |
LL |         /// `cfg=... and not `#[cfg_attr]\`
   |                                          +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:193:93
   |
LL |     /// Constructs a `TyKind::Error` type and registers a `span_delayed_bug` with the given `msg to
   |                                                                                             ^
   |
help: the closing backtick of an inline code may be missing
   |
LL |     /// Constructs a `TyKind::Error` type and registers a `span_delayed_bug` with the given `msg` to
   |                                                                                                 +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// Constructs a `TyKind::Error` type and registers a `span_delayed_bug` with the given \`msg to
   |                                                                                             +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:202:34
   |
LL |         /// in `nt_to_tokenstream`
   |                                  ^
   |
help: a previous inline code might be longer than expected
   |
LL |         /// if we parsed no predicates (e.g. `struct` Foo where {}
   |                                                     +
help: if you meant to use a literal backtick, escape it
   |
LL |         /// in `nt_to_tokenstream\`
   |                                  +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:208:62
   |
LL |     /// that `Option<Symbol>` only takes up 4 bytes, because `newtype_index! reserves
   |                                                              ^
   |
help: the closing backtick of an inline code may be missing
   |
LL |     /// that `Option<Symbol>` only takes up 4 bytes, because `newtype_index!` reserves
   |                                                                             +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// that `Option<Symbol>` only takes up 4 bytes, because \`newtype_index! reserves
   |                                                              +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:216:52
   |
LL |     /// also avoids the need to import `OpenOptions`.
   |                                                    ^
   |
help: a previous inline code might be longer than expected
   |
LL |     /// you can write `File::with_options().read(true).open("foo.txt")`. This
   |                                                                       +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// also avoids the need to import `OpenOptions\`.
   |                                                    +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:221:47
   |
LL |     /// `ChunkedBitSet`. Has no effect if `row` does not exist.
   |                                               ^
   |
help: a previous inline code might be longer than expected
   |
LL |     /// Subtracts `set` from `row`. `set` can be either `BitSet` or
   |                       +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// `ChunkedBitSet`. Has no effect if `row\` does not exist.
   |                                               +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:247:12
   |
LL |     /// RWU`s can get very large, so it uses a more compact representation.
   |            ^
   |
help: the opening backtick of an inline code may be missing
   |
LL |     /// `RWU`s can get very large, so it uses a more compact representation.
   |         +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// RWU\`s can get very large, so it uses a more compact representation.
   |            +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:254:15
   |
LL |     /// in `U2`.
   |               ^
   |
help: the opening backtick of a previous inline code may be missing
   |
LL |     /// `'?1` is in `U3` would be canonicalized to have `?0` in `U1` and `'?1`
   |                                                         +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// in `U2\`.
   |               +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:271:42
   |
LL |     /// because it contains `[type error]`. Yuck! (See issue #29857 for
   |                                          ^
   |
help: a previous inline code might be longer than expected
   |
LL |     /// as Trait>::Foo == $0`. Here, normalizing `<MyType<$0>` as
   |                                                              +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// because it contains `[type error]\`. Yuck! (See issue #29857 for
   |                                          +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:281:53
   |
LL |     /// well as the second instance of `A: AutoTrait`) to suppress
   |                                                     ^
   |
help: a previous inline code might be longer than expected
   |
LL |     /// encountered a problem (later on) with `A:` AutoTrait. So we
   |                                                  +
help: if you meant to use a literal backtick, escape it
   |
LL |     /// well as the second instance of `A: AutoTrait\`) to suppress
   |                                                     +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:291:40
   |
LL |     /// `'a` with `'b` and not `'static`. But it will have to do for
   |                                        ^
   |
   = help: the opening or closing backtick of an inline code may be missing
help: if you meant to use a literal backtick, escape it
   |
LL |     /// `'a` with `'b` and not `'static\`. But it will have to do for
   |                                        +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:300:54
   |
LL | /// `None`. Otherwise, it will return `Some(Dispatch)`.
   |                                                      ^
   |
help: the opening backtick of a previous inline code may be missing
   |
LL | /// The `Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
   |         +
help: if you meant to use a literal backtick, escape it
   |
LL | /// `None`. Otherwise, it will return `Some(Dispatch)\`.
   |                                                      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:304:13
   |
LL | /// or `None` if it isn't.
   |             ^
   |
   = help: the opening or closing backtick of an inline code may be missing
help: if you meant to use a literal backtick, escape it
   |
LL | /// or `None\` if it isn't.
   |             +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:308:14
   |
LL | /// `on_event` should be called.
   |              ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// Called before the filtered [`Layer`]'s [`on_event`], to determine if
   |                                       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// `on_event\` should be called.
   |              +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:313:29
   |
LL | /// [`rebuild_interest_cache`][rebuild] is called after the value of the max
   |                             ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// Therefore, if the `Filter` will change the value returned by this
   |                              +
help: if you meant to use a literal backtick, escape it
   |
LL | /// [`rebuild_interest_cache\`][rebuild] is called after the value of the max
   |                             +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:323:5
   |
LL | /     /// The Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
...  |
LL | |     /// [`rebuild_interest_cache`][rebuild] is called after the value of the max
LL | |     /// level changes.
   | |______________________^
   |
   = help: the opening backtick of a previous inline code may be missing
            change: The Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
           to this: The `Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
   = help: if you meant to use a literal backtick, escape it
            change: `None`. Otherwise, it will return `Some(Dispatch)`.
           to this: `None`. Otherwise, it will return `Some(Dispatch)\`.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:323:5
   |
LL | /     /// The Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
...  |
LL | |     /// [`rebuild_interest_cache`][rebuild] is called after the value of the max
LL | |     /// level changes.
   | |______________________^
   |
   = help: the opening or closing backtick of an inline code may be missing
   = help: if you meant to use a literal backtick, escape it
            change: or `None` if it isn't.
           to this: or `None\` if it isn't.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:323:5
   |
LL | /     /// The Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
...  |
LL | |     /// [`rebuild_interest_cache`][rebuild] is called after the value of the max
LL | |     /// level changes.
   | |______________________^
   |
   = help: a previous inline code might be longer than expected
            change: Called before the filtered [`Layer]'s [`on_event`], to determine if
           to this: Called before the filtered [`Layer`]'s [`on_event`], to determine if
   = help: if you meant to use a literal backtick, escape it
            change: `on_event` should be called.
           to this: `on_event\` should be called.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:323:5
   |
LL | /     /// The Subscriber` may be accessed by calling [`WeakDispatch::upgrade`],
...  |
LL | |     /// [`rebuild_interest_cache`][rebuild] is called after the value of the max
LL | |     /// level changes.
   | |______________________^
   |
   = help: a previous inline code might be longer than expected
            change: Therefore, if the `Filter will change the value returned by this
           to this: Therefore, if the `Filter` will change the value returned by this
   = help: if you meant to use a literal backtick, escape it
            change: [`rebuild_interest_cache`][rebuild] is called after the value of the max
           to this: [`rebuild_interest_cache\`][rebuild] is called after the value of the max

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:349:56
   |
LL |     /// instead and use [`CloneCounterObserver::counter`] to increment.
   |                                                        ^
   |
   = help: the opening or closing backtick of an inline code may be missing
help: if you meant to use a literal backtick, escape it
   |
LL |     /// instead and use [`CloneCounterObserver::counter\`] to increment.
   |                                                        +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:12:5
   |
---

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:26:6
   |
LL | /// [`link1]
   |      ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// [`link1`]
   |            +
help: if you meant to use a literal backtick, escape it
   |
LL | /// [\`link1]
   |      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:30:11
   |
LL | /// [link2`]
   |           ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// [`link2`]
   |      +
help: if you meant to use a literal backtick, escape it
   |
LL | /// [link2\`]
   |           +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:34:6
   |
LL | /// [`link_long](link_long)
   |      ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// [`link_long`](link_long)
   |                +
help: if you meant to use a literal backtick, escape it
   |
LL | /// [\`link_long](link_long)
   |      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:38:6
   |
LL | /// [`broken-link]
   |      ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// [`broken-link`]
   |                  +
help: if you meant to use a literal backtick, escape it
   |
LL | /// [\`broken-link]
   |      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:45:8
   |
LL | /// <x:`>
   |        ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// `<x:`>
   |     +
help: if you meant to use a literal backtick, escape it
   |
LL | /// <x:\`>
   |        +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:55:6
   |
---
LL | /// `foo(`
   |          +
help: if you meant to use a literal backtick, escape it
   |
LL | /// \`foo(
   |     +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:65:14
   |
LL | /// `foo `bar`
   |              ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// `foo` `bar`
   |         +
help: if you meant to use a literal backtick, escape it
   |
LL | /// `foo `bar\`
   |              +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:71:5
   |
---
LL | /// not paragraph`
   |                  +
help: if you meant to use a literal backtick, escape it
   |
LL | /// \`foo(
   |     +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:76:83
   |
LL | /// Addition is commutative, which means that add(a, b)` is the same as `add(b, a)`.
   |                                                                                   ^
   |
help: the opening backtick of a previous inline code may be missing
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   |                                               +
help: if you meant to use a literal backtick, escape it
   |
LL | /// Addition is commutative, which means that add(a, b)` is the same as `add(b, a)\`.
   |                                                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:80:51
   |
LL | /// or even to add a number `n` to 42 (`add(42, b)`)!
   |                                                   ^
   |
help: the opening backtick of a previous inline code may be missing
   |
LL | /// You could use this function to add 42 to a number `n` (`add(n, 42)`),
   |                                                            +
help: if you meant to use a literal backtick, escape it
   |
LL | /// or even to add a number `n` to 42 (`add(42, b)\`)!
   |                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:84:83
   |
LL | /// Addition is commutative, which means that `add(a, b) is the same as `add(b, a)`.
   |                                                                                   ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   |                                                         +
help: if you meant to use a literal backtick, escape it
   |
LL | /// Addition is commutative, which means that `add(a, b) is the same as `add(b, a)\`.
   |                                                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:88:51
   |
LL | /// or even to add a number `n` to 42 (`add(42, n)`)!
   |                                                   ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// You could use this function to add 42 to a number `n` (`add(n, 42)`),
   |                                                                       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// or even to add a number `n` to 42 (`add(42, n)\`)!
   |                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:92:83
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as add(b, a)`.
   |                                                                                   ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   |                                                                          +
help: if you meant to use a literal backtick, escape it
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as add(b, a)\`.
   |                                                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:96:50
   |
LL | /// or even to add a number `n` to 42 (add(42, n)`)!
   |                                                  ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// or even to add a number `n` to 42 (`add(42, n)`)!
   |                                        +
help: if you meant to use a literal backtick, escape it
   |
LL | /// or even to add a number `n` to 42 (add(42, n)\`)!
   |                                                  +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:100:74
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as `add(b, a).
   |                                                                          ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   |                                                                                    +
help: if you meant to use a literal backtick, escape it
   |
LL | /// Addition is commutative, which means that `add(a, b)` is the same as \`add(b, a).
   |                                                                          +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:104:51
   |
LL | /// or even to add a number `n` to 42 (`add(42, n)`)!
   |                                                   ^
   |
help: a previous inline code might be longer than expected
   |
LL | /// You could use this function to add 42 to a number `n` (`add(n, 42)`),
   |                                                                       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// or even to add a number `n` to 42 (`add(42, n)\`)!
   |                                                   +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:108:9
   |
LL | #[doc = "`"]
   |         ^^^
   |
   = help: the opening or closing backtick of an inline code may be missing
   = help: if you meant to use a literal backtick, escape it
            change: `
           to this: \`

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:115:9
   |
LL | #[doc = concat!("\\\\", "`")]
   |         ^^^^^^^^^^^^^^^^^^^^
   |
   = help: the opening backtick of an inline code may be missing
            change: \\`
           to this: `\\`
   = help: if you meant to use a literal backtick, escape it
            change: \\`
           to this: \\\`

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:119:9
   |
LL | #[doc = "Addition is commutative, which means that add(a, b)` is the same as `add(b, a)`."]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: the opening backtick of a previous inline code may be missing
            change: Addition is commutative, which means that add(a, b)` is the same as `add(b, a)`.
           to this: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   = help: if you meant to use a literal backtick, escape it
            change: Addition is commutative, which means that add(a, b)` is the same as `add(b, a)`.
           to this: Addition is commutative, which means that add(a, b)` is the same as `add(b, a)\`.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:123:9
   |
LL | #[doc = "Addition is commutative, which means that `add(a, b) is the same as `add(b, a)`."]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: a previous inline code might be longer than expected
            change: Addition is commutative, which means that `add(a, b) is the same as `add(b, a)`.
           to this: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   = help: if you meant to use a literal backtick, escape it
            change: Addition is commutative, which means that `add(a, b) is the same as `add(b, a)`.
           to this: Addition is commutative, which means that `add(a, b) is the same as `add(b, a)\`.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:127:9
   |
LL | #[doc = "Addition is commutative, which means that `add(a, b)` is the same as add(b, a)`."]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: the opening backtick of an inline code may be missing
            change: Addition is commutative, which means that `add(a, b)` is the same as add(b, a)`.
           to this: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   = help: if you meant to use a literal backtick, escape it
            change: Addition is commutative, which means that `add(a, b)` is the same as add(b, a)`.
           to this: Addition is commutative, which means that `add(a, b)` is the same as add(b, a)\`.

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:131:9
   |
LL | #[doc = "Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)."]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: the closing backtick of an inline code may be missing
            change: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a).
           to this: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a)`.
   = help: if you meant to use a literal backtick, escape it
            change: Addition is commutative, which means that `add(a, b)` is the same as `add(b, a).
           to this: Addition is commutative, which means that `add(a, b)` is the same as \`add(b, a).

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:136:5
   |
LL | /// `foo
---

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:140:7
   |
LL | /// # `(heading
   |       ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// # `(heading`
   |                +
help: if you meant to use a literal backtick, escape it
   |
LL | /// # \`(heading
   |       +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:142:17
   |
LL | /// ## heading2)`
   |                 ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// ## `heading2)`
   |        +
help: if you meant to use a literal backtick, escape it
   |
LL | /// ## heading2)\`
   |                 +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:145:11
   |
---

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:151:10
   |
LL | /// para)`(graph
   |          ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// `para)`(graph
   |     +
help: the closing backtick of an inline code may be missing
   |
LL | /// para)`(graph`
   |                 +
help: if you meant to use a literal backtick, escape it
   |
LL | /// para)\`(graph
   |          +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:154:10
   |
LL | /// para)`(graph2
   |          ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// `para)`(graph2
   |     +
help: the closing backtick of an inline code may be missing
   |
LL | /// para)`(graph2`
   |                  +
help: if you meant to use a literal backtick, escape it
   |
LL | /// para)\`(graph2
   |          +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:157:12
   |
LL | /// 1. foo)`
   |            ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// 1. `foo)`
   |        +
help: if you meant to use a literal backtick, escape it
   |
LL | /// 1. foo)\`
   |            +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:159:8
   |
LL | /// 2. `(bar
   |        ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// 2. `(bar`
   |             +
help: if you meant to use a literal backtick, escape it
   |
LL | /// 2. \`(bar
   |        +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:161:11
   |
LL | /// * baz)`
   |           ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// * `baz)`
   |       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// * baz)\`
   |           +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:163:7
   |
LL | /// * `(quux
   |       ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// * `(quux`
   |             +
help: if you meant to use a literal backtick, escape it
   |
LL | /// * \`(quux
   |       +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:166:5
   |
LL | /// `#![this_is_actually_an_image(and(not), an = "attribute")]
   |     ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// `#`![this_is_actually_an_image(and(not), an = "attribute")]
   |       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// \`#![this_is_actually_an_image(and(not), an = "attribute")]
   |     +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:169:62
   |
LL | /// #![this_is_actually_an_image(and(not), an = "attribute")]`
   |                                                              ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// `#![this_is_actually_an_image(and(not), an = "attribute")]`
   |     +
help: if you meant to use a literal backtick, escape it
   |
LL | /// #![this_is_actually_an_image(and(not), an = "attribute")]\`
   |                                                              +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:174:7
   |
LL | /// | `table( | )head` |
   |       ^
   |
help: the closing backtick of an inline code may be missing
   |
LL | /// | `table(` | )head` |
   |              +
help: if you meant to use a literal backtick, escape it
   |
LL | /// | \`table( | )head` |
   |       +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:174:22
   |
LL | /// | `table( | )head` |
   |                      ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// | `table( | `)head` |
   |                 +
help: if you meant to use a literal backtick, escape it
   |
LL | /// | `table( | )head\` |
   |                      +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:178:12
   |
LL | /// | table`( | )`body |
   |            ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// | `table`( | )`body |
   |       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// | table\`( | )`body |
   |            +

error: unescaped backtick
##[error]  --> /checkout/tests/rustdoc-ui/unescaped_backticks.rs:178:18
   |
LL | /// | table`( | )`body |
   |                  ^
   |
help: the opening backtick of an inline code may be missing
   |
LL | /// | table`( | `)`body |
   |                 +
help: the closing backtick of an inline code may be missing
   |
LL | /// | table`( | )`body` |
   |                       +
help: if you meant to use a literal backtick, escape it
   |
LL | /// | table`( | )\`body |
   |                  +

error: aborting due to 64 previous errors
------------------------------------------

```

</details>

--- a/compiler/rustc_resolve/src/rustdoc.rs
+++ b/compiler/rustc_resolve/src/rustdoc.rs
@@ -12,9 +12,13 @@ use rustc_data_structures::fx::FxIndexMap;
 use rustc_data_structures::unord::UnordSet;
 use rustc_middle::ty::TyCtxt;
 use rustc_span::def_id::DefId;
+use rustc_span::source_map::SourceMap;
 use rustc_span::{DUMMY_SP, InnerSpan, Span, Symbol, sym};
 use thin_vec::ThinVec;
 use tracing::{debug, trace};
+
+#[cfg(test)]
+mod tests;
 
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 pub enum DocFragmentKind {
@@ -532,9 +536,19 @@ pub fn source_span_for_markdown_range(
     md_range: &Range<usize>,
     fragments: &[DocFragment],
 ) -> Option<Span> {
+    let map = tcx.sess.source_map();
+    source_span_for_markdown_range_inner(map, markdown, md_range, fragments)
+}
+
+// inner function used for unit testing
+pub fn source_span_for_markdown_range_inner(
+    map: &SourceMap,
+    markdown: &str,
+    md_range: &Range<usize>,
+    fragments: &[DocFragment],
+) -> Option<Span> {
     use rustc_span::BytePos;
 
-    let map = tcx.sess.source_map();
     if let &[fragment] = &fragments
         && fragment.kind == DocFragmentKind::RawDoc
         && let Ok(snippet) = map.span_to_snippet(fragment.span)
@@ -570,7 +584,13 @@ pub fn source_span_for_markdown_range(
             {
                 // If there is either a match in a previous fragment, or
                 // multiple matches in this fragment, there is ambiguity.
-                if match_data.is_none() && !snippet[match_start + 1..].contains(pat) {
+                // the snippet cannot be zero-sized, because it matches
+                // the pattern, which is checked to not be zero sized.
+                if match_data.is_none()
+                    && !snippet.as_bytes()[match_start + 1..]
+                        .windows(pat.len())
+                        .any(|s| s == pat.as_bytes())
+                {
                     match_data = Some((i, match_start));
                 } else {
                     // Heirustic produced ambiguity, return nothing.

--- a/compiler/rustc_resolve/src/rustdoc/tests.rs
+++ b/compiler/rustc_resolve/src/rustdoc/tests.rs
@@ -1,0 +1,50 @@
+use std::path::PathBuf;
+
+use rustc_span::source_map::{FilePathMapping, SourceMap};
+use rustc_span::symbol::sym;
+use rustc_span::{BytePos, Span};
+
+use super::{DocFragment, DocFragmentKind, source_span_for_markdown_range_inner};
+
+#[test]
+fn single_backtick() {
+    let sm = SourceMap::new(FilePathMapping::empty());
+    sm.new_source_file(PathBuf::from("foo.rs").into(), r#"#[doc = "`"] fn foo() {}"#.to_string());
+    let span = source_span_for_markdown_range_inner(
+        &sm,
+        "`",
+        &(0..1),
+        &[DocFragment {
+            span: Span::with_root_ctxt(BytePos(8), BytePos(11)),
+            item_id: None,
+            kind: DocFragmentKind::RawDoc,
+            doc: sym::empty, // unused placeholder
+            indent: 0,
+        }],
+    )
+    .unwrap();
+    assert_eq!(span.lo(), BytePos(9));
+    assert_eq!(span.hi(), BytePos(10));
+}
+
+#[test]
+fn utf8() {
+    // regression test for https://github.com/rust-lang/rust/issues/141665
+    let sm = SourceMap::new(FilePathMapping::empty());
+    sm.new_source_file(PathBuf::from("foo.rs").into(), r#"#[doc = "⚠"] fn foo() {}"#.to_string());
+    let span = source_span_for_markdown_range_inner(
+        &sm,
+        "⚠",
+        &(0..3),
+        &[DocFragment {
+            span: Span::with_root_ctxt(BytePos(8), BytePos(14)),
+            item_id: None,
+            kind: DocFragmentKind::RawDoc,
+            doc: sym::empty, // unused placeholder
+            indent: 0,
+        }],
+    )
+    .unwrap();
+    assert_eq!(span.lo(), BytePos(9));
+    assert_eq!(span.hi(), BytePos(12));
+}

